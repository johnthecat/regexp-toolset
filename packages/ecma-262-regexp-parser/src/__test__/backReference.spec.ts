import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../index.js';

describe('Back Reference', () => {
  it('should parse back reference', () => {
    expect(parseRegexp(/(a)\1/)).toMatchSnapshot();
  });
});
