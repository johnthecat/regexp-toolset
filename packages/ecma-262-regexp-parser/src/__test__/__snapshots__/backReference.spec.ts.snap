// Vitest Snapshot v1

exports[`Back Reference > should parse back reference 1`] = `
{
  "body": {
    "end": 5,
    "group": {
      "body": {
        "charCode": 97,
        "end": 2,
        "kind": 6,
        "start": 2,
        "type": "simple",
        "value": "a",
      },
      "end": 3,
      "kind": 23,
      "specifier": null,
      "start": 1,
      "type": "capturing",
    },
    "kind": 25,
    "start": 1,
  },
  "end": 6,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
